console/services/market_app/app_upgrade.py

```diff
@@ -328,13 +328,18 @@ def _create_new_app(self):
 
         components = new_components + update_components
 
+        # component existing in the template.
+        tmpl_components = self._tmpl_components(components)
+        tmpl_component_ids = [cpt.component.component_id for cpt in tmpl_components]
+
         # create new component dependency from app_template
         new_component_deps = self._create_component_deps(components)
-        component_deps = self.ensure_component_deps(self.original_app, new_component_deps)
+        component_deps = self.ensure_component_deps(self.original_app, new_component_deps, tmpl_component_ids,
```

Comment:
Class methods don't have to pass class variables?

Reply:
fixed